resources/git-fork.md

```diff
@@ -0,0 +1,98 @@
+
+## What is a Fork?
+
+Forking is creating a new copy of the entire repository. This might sound like `git clone`, and behind the scenes it is. The difference is in where the new copy resides. A "clone" typically refers to a _local_ copy of some repository. A _fork_, in contrast, is a second *remote* repository distinct from the original that it copies.
+
+Historically, a _fork_ was created when a developer or team wanted to take a project in a new direction, or carry on an original mission where the core project undergoes some major shift. Some examples of this are NodeJS and IO.js, or when the licensing changes to `OpenOffice` by Sun led to the LibreOffice fork, and ultimately OpenOffice's demise. These classic situations can be referred to as **hard forks**, where the forked code is not meant to be integrated back into the original (upstream) codebase. What we deal with most often are so called **soft forks**, where we _do_ intend for our changes to be integrated back into the original repository, and the fork is a convenient system for creating an isolated workspace. From here on when we say "fork" we will be talking about soft forks, unless explicitly stated otherwise.
+
+## How do I make a fork?
+
+You can do this manually by cloning a repository locally and then pushing to a new remote repository in a place of your choosing (GitHub, GitLab, BitBucket, etc). You would do so like this:
+```bash
+$> git clone git@github.com:some-account/some-repo.git
+$> git remote add myfork git@github.com:fork-account/fork-repo.git
+$> git push myfork master
+```
+Note: more about [managing `remote`s](#dealing-with-multiple-remotes) below.
+
+GitHub and most other git-based version control services have forking built in as a feature that will automatically create a new repository under your account and copy the source repository into your new repository.
+
+On GitHub, for example, this can be accomplished simply by clicking the "Fork" button in the top right corner, just under your account avatar. See here:
+![Forking on GitHub](images/github-fork.png)
+
+## Why would we want a fork?
+### Why not just branch?
+
+So far we have mostly discussed the reasons one may want or need to create a _hard_ fork of some project, where we intend on making permanently incompatible changes, but we have not justified our more typical (soft) fork situation. As you recall, this is for work which we want to eventually merge back into the main repository. Some reasons for doing this are:
+
+* **User workspaces:** when projects involve a large number of developers (imagine a popular open source or large enterprise project), if each developer were pushing branches for their work the repository would become cluttered with many branches mostly of interest to only one person. When you are synchronizing your local instance, you have many more branches listed than you need because you may see branches for everybody's work.
```

Comment:
comma here? `if each developer were pushing branches for their work, the repository`

Reply:
I think maybe this first point can be reworded slightly. How about something like this:

`some projects involve a large number of developers (imagine a popular open source or large enterprise project) and if every developer pushes multiple branches of their work to the same repository, that repository can become very cluttered. When you synchronize your local instance in those cases, you will have many more branches listed than you need or care about as you'll see branches for everybody's work.`